### PR TITLE
add runtime display  for nerdctl ps

### PR DESCRIPTION
**Pull request by ningmingxiao**

### CODE REVIEW

**Comment by ningmingxiao** [comment]
[root@centos7 _output]# ctr c ls
CONTAINER                                                           IMAGE                               RUNTIME
d3fca4e7912c7006ec248a822b4c0b88eb2dea90c4cfc085bc67879d87c3f7c6    docker.io/library/busybox:latest    io.containerd.runc.v2
nerdctl should support  it

File: cmd/nerdctl/ps.go [hunk 1]
```diff
@@ -70,14 +70,15 @@ func psAction(cmd *cobra.Command, args []string) error {
...
-	Status    string
+	Command     string
+	CreatedAt   string
+	ID          string
+	Image       string
+	Platform    string // nerdctl extension
+	Names       string
+	Ports       string
+	Status      string
+	RunTimeName string
```
Comment (AkihiroSuda) [inline cmd/nerdctl/ps.go:81]: `RuntimeName string // nerdctl extension`

File: cmd/nerdctl/ps.go [hunk 2]
```diff
@@ -106,7 +107,7 @@ func printContainers(ctx context.Context, cmd *cobra.Command, containers []conta
 	case "", "table":
 		w = tabwriter.NewWriter(os.Stdout, 4, 8, 4, ' ', 0)
 		if !quiet {
-			fmt.Fprintln(w, "CONTAINER ID\tIMAGE\tPLATFORM\tCOMMAND\tCREATED\tSTATUS\tPORTS\tNAMES")
+			fmt.Fprintln(w, "CONTAINER ID\tIMAGE\tPLATFORM\tCOMMAND\tCREATED\tSTATUS\tPORTS\tNAMES\tRUNTIME")
```
Comment (AkihiroSuda) [inline cmd/nerdctl/ps.go:110]: I think we should add `RUNTIME` field only when we have sufficient TTY width (and when we don't have TTY)


https://pkg.go.dev/golang.org/x/term#GetSize

File: cmd/nerdctl/ps.go [hunk 3]
```diff
@@ -78,6 +78,7 @@ type containerPrintable struct {
 	Names     string
 	Ports     string
 	Status    string
+	RunTime   string
```
Comment (AkihiroSuda) [inline cmd/nerdctl/ps.go:81]:
```suggestion
	Runtime string // nerdctl extension`
```

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]

File: cmd/nerdctl/ps.go [hunk 4]
```diff
@@ -78,6 +78,7 @@ type containerPrintable struct {
 	Names     string
 	Ports     string
 	Status    string
+	RunTime   string // nerdctl extension
```
Comment (AkihiroSuda) [inline cmd/nerdctl/ps.go:81]:
```suggestion
	Runtime   string // nerdctl extension
```

File: cmd/nerdctl/ps.go (hunk 2 above)
Comment (AkihiroSuda) [reply cmd/nerdctl/ps.go:110]: Or maybe only when `--no-trunc` is specified

Comment (ningmingxiao) [reply cmd/nerdctl/ps.go:110]: truncate means  "Don't truncate output" which isn't the right word to use here

Comment (AkihiroSuda) [reply cmd/nerdctl/ps.go:110]: What about `--format wide`

Comment (ningmingxiao) [reply cmd/nerdctl/ps.go:110]: format means make output  according to some standards,
maybe -o wide is ok, like kubectl get node -o wide?

Comment (ningmingxiao) [reply cmd/nerdctl/ps.go:110]: but -o wide maybe conflict with  no-trunc

Comment (AkihiroSuda) [reply cmd/nerdctl/ps.go:110]: I don't think we should have both `-o` and `--format`.

Comment (ningmingxiao) [reply cmd/nerdctl/ps.go:110]: maybe  --format wide is right word

**Comment by ningmingxiao** [comment]
I move platform  and runtime to --fortmat wide,bracause  platform's value often null

**Review by AkihiroSuda — APPROVED** [review]
Thanks, added a follow-up commit to approve this

**Pull request merged**